### PR TITLE
feat: add "Clear all" action to un-apply eval tags and delete runs

### PR DESCRIPTION
**Pull request by SmittieC**
### Product Description
Adds a **Clear all** button to the evaluation runs page. It lets users with delete permission wipe a config's run history in one action — and, crucially, also removes the tags those runs applied to their targets (chats / chat messages), so the targets are returned to the state they were in before the evaluator ran. Tags a person added by hand are left untouched.

### Technical Description
- New `ClearEvaluationRuns` view (`POST .../runs/clear/`), gated on the `evaluations.delete_evaluationrun` permission. It runs in a single transaction: un-apply tags, then delete the runs.
- New `remove_applied_tags_for_runs` helper in `apps/evaluations/tagging.py`. It walks each run's `AppliedTag` audit, resolves each result's target per the evaluator's `evaluation_mode`, and deletes only the eval-created `CustomTaggedItem` links (`user__isnull=True`). It must run before the runs are deleted, since it reads the audit to know which tags to un-apply.
- Template: a "Clear all" button on the runs home page, shown only when the user has delete permission, with an `alertify` confirmation prompt.
- Tests covering the helper (message/session targets, manual-tag preservation, multi-run aggregation, no-audit and none-target edge cases) and the view (clears + deletes, preserves manual tags, scoped to the config, permission enforcement, button visibility).

### Migrations
- [x] The migrations are backwards compatible

(No migrations in this PR.)

### Demo
https://www.loom.com/share/0476a268c41d4cff870ac90003f79e24

### Docs and Changelog
- [x] This PR requires docs/changelog update

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/evaluations/tagging.py [hunk 1]
```diff
@@ -206,3 +206,43 @@ def reverse_stale_tags(run: EvaluationRun) -> None:
...
+    content_type: ContentType | None = None
+    pairs: set[tuple[int, int]] = set()
+    for applied_tag in applied_tags:
+        result = applied_tag.evaluation_result
+        target = resolve_target(result.evaluator, result.message)
+        if target is None:
+            continue
+        if content_type is None:
+            content_type = ContentType.objects.get_for_model(type(target))
+        pairs.add((target.pk, applied_tag.tag_id))
```
[the remark this answers is not in the document]
Comment (SmittieC) [reply apps/evaluations/tagging.py:240]: `evaluation_mode` cannot be changed

**Comment by SmittieC** [comment]
@snopoke this replaces https://github.com/dimagi/open-chat-studio/pull/3517

File: templates/evaluations/evaluation_runs_home.html [hunk 2]
```diff
@@ -38,6 +38,22 @@ <h1 class="text-2xl font-bold">{{ title }}</h1>
           <i class="fa-solid fa-play"></i>
           Run Evaluation
         </a>
+        {% if perms.evaluations.delete_evaluationrun %}
+          <button id="clear-all-runs-button"
```
Comment (snopoke) [inline templates/evaluations/evaluation_runs_home.html:42]: this could use some progress indication - disable + spinner?

**Review by snopoke — COMMENTED** [review]
one comment but looks good.
